rest/data/commit_queue.go

```diff
@@ -377,6 +351,72 @@ func EnqueuePRToCommitQueue(ctx context.Context, env evergreen.Environment, sc C
 	return apiPatch, nil
 }
 
+// getPRAndCheckBase gets the Github PR and verifies that base and base ref is set
+func getPRAndCheckBase(ctx context.Context, sc Connector, info commitqueue.EnqueuePRInfo) (*github.PullRequest, error) {
+	pr, err := sc.GetGitHubPR(ctx, info.Owner, info.Repo, info.PR)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting PR from GitHub API")
+	}
+	if pr == nil || pr.Base == nil || pr.Base.Ref == nil {
+		return nil, errors.New("PR contains no base branch label")
+	}
+	return pr, nil
+}
+
+// getPRAndCheckMergeable gets the Github PR, verifies base, and verifies that the PR is mergeable.
+// Attempts to refresh the status if the PR is marked as blocked but the patch is successful.
+func getPRAndCheckMergeable(ctx context.Context, env evergreen.Environment, sc Connector, info commitqueue.EnqueuePRInfo) (*github.PullRequest, error) {
+	pr, err := getPRAndCheckBase(ctx, sc, info)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the PR is blocked and the patch is successful, refresh status, and re-check PR.
+	if pr.GetMergeableState() == thirdparty.GithubPRBlocked {
+		p, err := patch.FindLatestGithubPRPatch(info.Owner, info.Repo, info.PR)
+		if err != nil {
+			// Not required that such a patch exists, since the commit queue can be enabled without PR checks, so we log at debugging.
+			grip.Debug(message.WrapError(err, message.Fields{
+				"message": "couldn't find latest PR patch when enqueuing PR",
+				"ticket":  "EVG-19098",
+				"owner":   info.Owner,
+				"repo":    info.Repo,
+				"pr":      info.PR,
+			}))
+		} else if p != nil && p.Version != "" && p.Status == evergreen.PatchSucceeded {
+			refreshJob := units.NewGithubStatusRefreshJob(p)
+			refreshJob.Run(ctx)
```

Comment:
Worth it to add a `"ticket": "EVG-19098"` debug log here as well for traceability?

Reply:
I actually meant to remove that elsewhere, it was just for my personal debugging. Imo tickets should only ever be used in debugging if the logging is being introduced to aid in a future or the same ticket (with the intention of removing it after the investigation is done), otherwise it's just noise.